pkg/operator/controller/configurable-route/controller_test.go

```diff
@@ -0,0 +1,899 @@
+				if err := r.client.Create(context.Background(), tc.ingressConfig); err != nil {
+					t.Errorf("error creating ingress config: %v", err)
+				}
+			}
+
+			if _, err := r.Reconcile(context.Background(), tc.request); err == nil && tc.expectError {
+				t.Errorf("expected error, got no error")
+			} else if err != nil && !tc.expectError {
+				t.Errorf("did not expected error: %v", err)
+			} else {
+				actualRoles := &rbacv1.RoleList{}
+				if err := r.client.List(context.Background(), actualRoles); err == nil {
+					for _, expectedRole := range tc.expectedRoles.Items {
+						// Find the actual role that we should compare status with this expected role
+						var actualRole rbacv1.Role
+						for _, role := range actualRoles.Items {
+							// TRICKY: Because ensureServiceCertKeyPairSecretRole *sometimes* generates a name for the
+							//         role, we need to compare both name and generated name.
+							if role.GenerateName == expectedRole.GenerateName || role.Name == expectedRole.Name {
+								actualRole = role
+								expectedRole.Name = role.Name
+							}
+						}
+						if assert.NotEmptyf(t, actualRole, fmt.Sprintf("Could not find expected role: %s/%s", expectedRole.Namespace, expectedRole.Name)) {
+							// Value that we want to ignore in the comparison.
+							actualRole.ResourceVersion = ""
+							actualRole.GenerateName = ""
+							expectedRole.GenerateName = ""
+							assert.Equal(t, expectedRole, actualRole)
+						}
+					}
+				} else {
+					t.Errorf("error retrieving roles from client: %v", err)
+				}
+				actualRoleBindings := &rbacv1.RoleBindingList{}
+				if err := r.client.List(context.Background(), actualRoleBindings); err == nil {
+					for _, expectedRoleBinding := range tc.expectedRoleBindings.Items {
+						// Find the actual role that we should compare status with this expected role
+						var actualRoleBinding rbacv1.RoleBinding
+						for _, roleBinding := range actualRoleBindings.Items {
+							// TRICKY: RoleBinding will have generated name, so we will look for the prefix.
+							if strings.HasPrefix(roleBinding.Name, expectedRoleBinding.Name+"-") {
+								actualRoleBinding = roleBinding
+								expectedRoleBinding.Name = actualRoleBinding.Name
+								expectedRoleBinding.RoleRef.Name = actualRoleBinding.Name
+								expectedRoleBinding.OwnerReferences[0].Name = actualRoleBinding.Name
+							}
+						}
+						if assert.NotEmptyf(t, actualRoleBinding, fmt.Sprintf("Could not find expected rolebinding: %s/%s", expectedRoleBinding.Namespace, expectedRoleBinding.Name)) {
+							// Value that we want to ignore in the comparison.
+							expectedRoleBinding.ResourceVersion = ""
+							assert.Equal(t, expectedRoleBinding, actualRoleBinding)
+						}
+					}
+				} else {
+					t.Errorf("error retrieving roles from client: %v", err)
+				}
+			}
+		})
+	}
+}
+
+// Test_intersectingComponentRoutes verifies intersectingComponentRoutes
+func Test_intersectingComponentRoutes(t *testing.T) {
```

Comment:
This function is already a part of Reconcile.  Does it give us additional value to unit test it outside of a unit test for Reconcile?

Reply:
We talked a bit in slack about this. `intersectingComponentRoutes` does provide additional value through testing edge cases with component routes. It tests for invalid component routes. I think these are good reasons to unit test and don't think we should remove these test cases.

However, I simply just moved the tests cases to the Reconcile function, since it was just as easy to test them there and removed this unit test. Let me know if you disagree.